[Docs] MessagePack IDL, Pydantic Support, and Attribute Access (#6022)
flyteorg/flyte: docs/user_guide/data_types_and_io/pydantic_basemodel.md

(pydantic_basemodel)=

# Pydantic BaseModel

```{eval-rst}
.. tags:: Basic
```

`flytekit` version >=1.14 supports natively the `JSON` format that Pydantic `BaseModel` produces,  enhancing the 

> interoperability of Pydantic BaseModels with the Flyte type system.

Reviewer: no need to insert new line

:::{important}
Pydantic BaseModel V2 only works when you are using flytekit version >= v1.14.0.
:::

With the 1.14 release, `flytekit` adopted `MessagePack` as the serialization format for Pydantic `BaseModel`, 
overcoming a major limitation of serialization into a JSON string within a Protobuf `struct` datatype like the previous versions do:

> 

Reviewer: ```suggestion
```

> to store `int` types, Protobuf's `struct` converts them to `float`, forcing users to write boilerplate code to work around this issue.

Reviewer: I think new line after a colon (unless a list) looks strange

> By default, `flytekit >= 1.14` will produce `msgpack` bytes literals when serializing dataclasses, preserving the types defined in your `BaseModel` class.

Reviewer: ```suggestion
```

:::{important}

> If you're serializing dataclasses using `flytekit` version >= v1.14.0 and you want to produce Protobuf `struct literal` instead, you can set environment variable `FLYTE_USE_OLD_DC_FORMAT` to `true`.

Reviewer: ```suggestion
By default, `flytekit >= 1.14` will produce `msgpack` bytes literals when serializing, preserving the types defined in your `BaseModel` class.
If you're serializing `BaseModel` using `flytekit` version >= v1.14.0 and you want to produce Protobuf `struct literal` instead, you can set environment variable `FLYTE_USE_OLD_DC_FORMAT` to `true`.
```

Consolidating on the `important` block

(pydantic_basemodel)=

# Pydantic BaseModel

```{eval-rst}
.. tags:: Basic
```

> When you have multiple values that you want to send across Flyte entities, and you want them to have, you can use a `pydantic.BaseModel`.

Reviewer: ```suggestion
`flytekit` version >=1.14 supports natively the `JSON` format that Pydantic `BaseModel` produces,  enhancing the interoperability of Pydantic schemas with the Flyte type system.
```

:::{important}
Pydantic BaseModel V2 only works when you are using flytekit version >= v1.14.0.
:::

> :::{important}

Reviewer: ```suggestion
With the 1.14 release, `flytekit` adopted `MessagePack` as the serialization format for Pydantic `BaseModel`, overcoming a major limitation of serialization into a JSON string within a Protobuf `struct` datatype like the previous versions do: to store `int` types, Protobuf's `struct` converts them to `float`, forcing users to write boilerplate code to workaround this issue. By default, `flytekit >= 1.14` will produce `msgpack` bytes literals when serializing dataclasses, preserving the types defined in your `BaseModel` class.
:::{important}
If you're serializing dataclasses using `flytekit` version >= v1.14.0 and you want to produce Protobuf `struct literal` instead, you can set environment variable `FLYTE_USE_OLD_DC_FORMAT` to `true`.
```

> If you're using Flytekit version >= v1.14.0 and you want to produce protobuf struct literal for Pydantic BaseModels,

Reviewer: ```suggestion

```

> you can set environment variable  `FLYTE_USE_OLD_DC_FORMAT` to `true`.

Reviewer: ```suggestion
```
